### PR TITLE
Add support for tun/tap configuration

### PR DESCRIPTION
**Pull request by cfconrad**
## Problem

Add support for tun/tap configuration

Fixes: #30

### CODE REVIEW

File: rust/migrate-wicked/src/tuntap.rs [hunk 1]
```diff
@@ -0,0 +1,39 @@
+use agama_server::network::model;
+use serde::{Deserialize, Serialize};
+use serde_with::skip_serializing_none;
+
+#[skip_serializing_none]
+#[derive(Debug, PartialEq, Default, Serialize, Deserialize)]
+#[serde(default)]
+pub struct Tun {
+    pub owner: Option<u32>,
+    pub group: Option<u32>,
```
Comment (jcronenberg) [inline rust/migrate-wicked/src/tuntap.rs:10]: Are these actually optional? When I omit e.g. `TUNNEL_SET_OWNER=` it seems to still populate `owner` to `0`

File: rust/migrate-wicked/src/tuntap.rs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+#[skip_serializing_none]
+#[derive(Debug, PartialEq, Default, Serialize, Deserialize)]
+#[serde(default)]
+pub struct Tap {
+    pub owner: Option<u32>,
+    pub group: Option<u32>,
```
Comment (jcronenberg) [inline rust/migrate-wicked/src/tuntap.rs:18]: Same here

File: rust/migrate-wicked/src/tuntap.rs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        })
+    }
+}
+
+impl From<&Tap> for model::ConnectionConfig {
+    fn from(tuntap: &Tap) -> model::ConnectionConfig {
+        model::ConnectionConfig::Tun(model::TunConfig {
+            mode: model::TunMode::Tap,
+            group: tuntap.group.map(|v| v.to_string()),
+            owner: tuntap.owner.map(|v| v.to_string()),
```
Comment (jcronenberg) [inline rust/migrate-wicked/src/tuntap.rs:36]: Just wondering, does this work? I can see in the NM docs that it requires e.g. for owner `The user ID which will own the device`. Is `ID` in this case really a u32 in string form or the actual name of the group and owner?

File: rust/migrate-wicked/src/tuntap.rs (hunk 1 above)
Comment (cfconrad) [reply rust/migrate-wicked/src/tuntap.rs:10]: Hm, you are right. On converting ifcfg to wicked always write the owner and group.

But xml could be used by itself with wicked.
```bash
tw-1:/etc/wicked/ifconfig # wicked ifup tun9
tun9            up
tw-1:/etc/wicked/ifconfig # wicked show-config tun9
<interface origin="wicked:xml:/etc/wicked/ifconfig/tun9.xml">
  <name>tun9</name>
  <control>
    <mode>boot</mode>
  </control>
  <firewall/>
  <tun>
    <group>2</group>
  </tun>
  <link/>
</interface>
tw-1:/etc/wicked/ifconfig # cat /etc/wicked/ifconfig/tun9.xml 
<interface >
  <name>tun9</name>
  <control>
    <mode>boot</mode>
  </control>
  <firewall/>
  <tun>
    <group>2</group>
  </tun>
  <link/>
</interface>
````

And such config get's applied to NetworkManager like this:
```
[connection]
id=tun9
uuid=4d7d662c-afa1-4a2f-ad3e-34d92eb6c105
type=tun
interface-name=tun9

[tun]
group=0

[match]

[ipv4]
method=disabled

[ipv6]
addr-gen-mode=default
method=disabled

[proxy]
```

Comment (jcronenberg) [reply rust/migrate-wicked/src/tuntap.rs:10]: Ah, I see

File: rust/migrate-wicked/src/tuntap.rs (hunk 3 above)
Comment (cfconrad) [reply rust/migrate-wicked/src/tuntap.rs:36]: I would say  yes, see: https://gitlab.freedesktop.org/NetworkManager/NetworkManager/blob/main/src/libnm-core-impl/nm-setting-tun.c#L169

Comment (jcronenberg) [reply rust/migrate-wicked/src/tuntap.rs:36]: You are right, seems like it is. Weird that it is implemented as a string in dbus.

**Review by jcronenberg — APPROVED** [review]

**Pull request merged**